pkg/dyninst/gosym/symtab.go

```diff
@@ -0,0 +1,1187 @@
+
+			offsetBase := inlinedCallSize * int(*lastCall.Index)
+
+			// Get funcid
+			if offsetBase+funcIDOffset < len(inlineTree) {
+				funcID := inlineTree[offsetBase+funcIDOffset]
+				lastCall.FuncID = &funcID
+			}
+
+			// Get parent PC
+			if offsetBase+parentPCOffset+4 <= len(inlineTree) {
+				parentPC := binary.LittleEndian.Uint32(inlineTree[offsetBase+parentPCOffset:])
+				nextPC = *entryPC + uint64(parentPC)
+			} else {
+				break
+			}
+
+			// Get function name
+			if offsetBase+functionNameOffOffset+4 <= len(inlineTree) {
+				nameOff := binary.LittleEndian.Uint32(inlineTree[offsetBase+functionNameOffOffset:])
+				if name := f.lt.funcName(nameOff); name != nil {
+					lastCall.Function = name
+				}
+			}
+		} else {
+			nextPC = pc
+		}
+
+		// Get inline tree index for this PC
+		index := f.pcValue(pcdataInlTreeIndex, nextPC)
+		var indexPtr *uint32
+		if index != nil && *index >= 0 {
+			idx := uint32(*index)
+			indexPtr = &idx
+		}
+
+		if indexPtr == nil && len(inlinedCalls) == 0 {
+			return nil
+		}
+
+		inlinedCalls = append(inlinedCalls, inlinedCall{
+			PC:       nextPC,
+			Index:    indexPtr,
+			Function: nil,
+			FuncID:   nil,
+		})
+	}
+
+	if len(inlinedCalls) == 0 {
+		return nil
+	}
+
+	return inlinedCalls
+}
+
+// funcTab represents the function table portion of lineTable.
+type funcTab lineTable
+
+func (ft *funcTab) count() uint32 {
+	return ft.nfunctab
+}
+
+func (ft *funcTab) pc(i uint32) (uint64, error) {
+	if i >= ft.nfunctab {
+		return 0, fmt.Errorf("function index out of range")
+	}
+
+	fieldSize := functabFieldSize(ft.ptrSize, ft.version)
+	offset := ft.functab[0] + int(2*i)*fieldSize
+
+	if offset+fieldSize > len(ft.data) {
+		return 0, fmt.Errorf("function table entry out of bounds")
+	}
+
+	var pc uint64
+	if fieldSize == 8 {
+		pc = binary.LittleEndian.Uint64(ft.data[offset:])
+	} else if fieldSize == 4 {
+		pc = uint64(binary.LittleEndian.Uint32(ft.data[offset:]))
+	} else {
+		return 0, fmt.Errorf("unexpected field size: %d", fieldSize)
+	}
+
+	// For ver118/120, add text_start for relocation
+	switch ft.version {
+	case ver118, ver120:
+		pc += ft.textRange[0]
+	}
+
+	return pc, nil
+}
+
+func (ft *funcTab) funcOff(i uint32) (uint64, error) {
+	if i >= ft.nfunctab {
+		return 0, fmt.Errorf("function index out of range")
+	}
+
+	fieldSize := functabFieldSize(ft.ptrSize, ft.version)
+	offset := ft.functab[0] + int(2*i+1)*fieldSize
+
+	if offset+fieldSize > len(ft.data) {
+		return 0, fmt.Errorf("function offset out of bounds")
+	}
+
+	var funcOff uint64
+	if fieldSize == 8 {
+		funcOff = binary.LittleEndian.Uint64(ft.data[offset:])
+	} else if fieldSize == 4 {
+		funcOff = uint64(binary.LittleEndian.Uint32(ft.data[offset:]))
+	} else {
+		return 0, fmt.Errorf("unexpected field size: %d", fieldSize)
+	}
+
+	return funcOff, nil
+}
+
+// funcInfo represents single function data in the pclntab.
+type funcInfo struct {
+	lt   *lineTable
+	data []byte
+}
+
+func (fi *funcInfo) field(n uint32) *uint32 {
```

Comment:
nit: I think this pointer to uint32 is excessive -- let's just do `(uint32, bool)`